storage/src/state/operator.rs

```diff
@@ -0,0 +1,187 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::storage::{DataMap, Map, Storage};
+use snarkvm::dpc::prelude::*;
+
+use anyhow::{anyhow, Result};
+use std::{collections::HashMap, path::Path};
+
+#[derive(Debug)]
+pub struct OperatorState<N: Network> {
+    shares: SharesState<N>,
+}
+
+impl<N: Network> OperatorState<N> {
+    ///
+    /// Opens a new writable instance of `OperatorState` from the given storage path.
+    ///
+    pub fn open_writer<S: Storage, P: AsRef<Path>>(path: P) -> Result<Self> {
+        // Open storage.
+        let context = N::NETWORK_ID;
+        let is_read_only = false;
+        let storage = S::open(path, context, is_read_only)?;
+
+        // Initialize the operator.
+        let operator = Self {
+            shares: SharesState::open(storage)?,
+        };
+
+        info!("Operator successfully initialized");
+        Ok(operator)
+    }
+
+    /// Returns all the shares in storage.
+    pub fn to_shares(&self) -> Vec<(u32, HashMap<Address<N>, u64>)> {
```

Comment:
Might be nice to introduce a type alias or a struct here?

Reply:
It won't really work that well cause we need it to be split into two different types for the `DataMap`. I can wrap the hashmap, but I don't really see the need..